src/webgpu/api/operation/sampling/anisotropy.spec.ts

```diff
@@ -6,14 +6,506 @@ Note that anisotropic filtering is never guaranteed to occur, but we might be ab
+  }
+
+  checkCheckerBoardResults(results: Uint8Array[]) {
+    this.eventualAsyncExpectation(async niceStack => {
+      if (true === this.compareUint8ArrayResult(results[0], results[1])) {
+        niceStack.message =
+          'Render results with sampler.maxAnisotropy being 1 and 16 should be different.';
+        this.rec.expectationFailed(niceStack);
+      }
+      if (false === this.compareUint8ArrayResult(results[1], results[2])) {
+        niceStack.message =
+          'Render results with sampler.maxAnisotropy being 16 and 1024 should be the same.';
+        this.rec.expectationFailed(niceStack);
+      }
+    });
+  }
+
+  copyRenderTargetToBuffer(rt: GPUTexture): GPUBuffer {
+    const byteLength = kRTSize * kBytesPerRow;
+    const buffer = this.device.createBuffer({
+      size: byteLength,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    const commandEncoder = this.device.createCommandEncoder();
+    commandEncoder.copyTextureToBuffer(
+      { texture: rt, mipLevel: 0, origin: [0, 0, 0] },
+      { buffer, bytesPerRow: kBytesPerRow, rowsPerImage: kRTSize },
+      { width: kRTSize, height: kRTSize, depth: 1 }
+    );
+    this.queue.submit([commandEncoder.finish()]);
+
+    return buffer;
+  }
+
+  private pipeline: GPURenderPipeline | undefined;
+  private uniformBuffer: GPUBuffer | undefined;
+  private vertexBuffer: GPUBuffer | undefined;
+  setup(): void {
+    this.pipeline = this.device.createRenderPipeline({
+      vertexStage: {
+        module: this.device.createShaderModule({
+          code: `
+            [[block]] struct Uniforms {
+                [[offset(0)]] matrix : mat4x4<f32>;
+            };
+
+            [[location(0)]] var<in> position : vec4<f32>;
+            [[location(1)]] var<in> uv : vec2<f32>;
+
+            [[set(0), binding(2)]] var<uniform> uniforms : Uniforms;
+
+            [[builtin(position)]] var<out> Position : vec4<f32>;
+            [[location(0)]] var<out> fragUV : vec2<f32>;
+
+            [[stage(vertex)]] fn main() -> void {
+                fragUV = uv;
+                Position = uniforms.matrix * position;
+            }
+            `,
+        }),
+        entryPoint: 'main',
+      },
+      fragmentStage: {
+        module: this.device.createShaderModule({
+          code: `
+            [[set(0), binding(0)]] var<uniform_constant> sampler0 : sampler;
+            [[set(0), binding(1)]] var<uniform_constant> texture0 : texture_2d<f32>;
+
+            [[builtin(frag_coord)]] var<in> FragCoord : vec4<f32>;
+
+            [[location(0)]] var<in> fragUV: vec2<f32>;
+
+            [[location(0)]] var<out> fragColor : vec4<f32>;
+
+            [[stage(fragment)]] fn main() -> void {
+                fragColor = textureSample(texture0, sampler0, fragUV);
+            }
+            `,
+        }),
+        entryPoint: 'main',
+      },
+      primitiveTopology: 'triangle-list',
+      colorStates: [{ format: 'rgba8unorm' }],
+      vertexState: {
```

Comment:
Consider using something like this:
https://github.com/gpuweb/cts/blob/main/src/webgpu/api/operation/rendering/basic.spec.ts#L65
instead of using vertex buffers. Helps cut down the size of the test.

You can also hardcode the matrix into a const in the shader. Optionally you could even pre-multiply the matrix into the position data so no matrix is needed.